pkg/controller/azvolume.go

```diff
@@ -153,15 +165,21 @@ func (r *reconcileAzVolume) UpdateStatus(ctx context.Context, volumeName string,
 
 	if isDeleted {
 		if err := r.DeleteFinalizer(ctx, volumeName); err != nil {
-			klog.Errorf("failed to delete finalizer %s for azVolume %s: %v", AzVolumeFinalizer, azVolume.Name, err)
+			klog.Errorf("failed to delete finalizer %s for azVolume %s: %v", azureutils.AzVolumeFinalizer, azVolume.Name, err)
 			return err
 		}
 		return nil
 	}
 
 	updated := azVolume.DeepCopy()
 	if updated.Status == nil {
-		updated.Status = &v1alpha1.AzVolumeStatus{}
+		updated.Status = &v1alpha1.AzVolumeStatus{
+			ResponseObject: status,
+		}
+	} else {
+		// Updating status after update operation
+		updated.Status.ResponseObject.CapacityBytes = status.CapacityBytes
+		updated.Status.ResponseObject.NodeExpansionRequired = status.NodeExpansionRequired
```

Comment:
If NodeExpansionRequired, do we need to make another call?

Reply:
I see, is that something that needs to occur in the AzVolume reconciler eventually? Or will it be separate?

Reply:
Hmmm. Yes, you're right, but the volume expansion scenario doesn't seem to be covered in the design document, but it looks like @mukhoakash plumbed the right support for this in the `AzVolume` controller in this PR.